libs/core/src/main/java/org/elasticsearch/core/Strings.java

```diff
@@ -21,8 +21,14 @@ public class Strings {
      *
      * This method calls {@link String#format(Locale, String, Object...)}
      * with Locale.ROOT
+     * If format is incorrect the function will return format without populating
+     * its variable placeholders.
      */
     public static String format(String format, Object... args) {
-        return String.format(Locale.ROOT, format, args);
+        try {
+            return String.format(Locale.ROOT, format, args);
+        } catch (Throwable e) {
```

Comment:
Why throwable? This should probably be just Exception. Let OOM/etc propagate.

Reply:
Exception is definitely better